digits_test.go

```diff
@@ -6,251 +6,341 @@ import (
+			strOut:     "80,800,000.99",
+			head:       "",
+			core:       "80,800,000.9",
+			tail:       "9",
+		},
 	},
 }
 
-func act() []*digits.Expression {
-	var suts = []*digits.Expression{}
-	for _, s := range stimuli {
-		d, err := digits.New(s.p, s.v, s.g, s.d)
-		if err != nil {
-			panic(err)
-		}
-		suts = append(suts, d)
+func process(stimulus Stimulus) *digits.Expression {
+	d, err := digits.New(stimulus.precision, stimulus.value, stimulus.groupSeparator, stimulus.fractionalPrecision)
+	if err != nil {
+		panic(err)
 	}
-	return suts
+	return d
 }
 func Test_SigFigs(t *testing.T) {
-	var suts = act()
-	expectations := []string{
-		"0.",
-		"0.02",
-		"0.02",
-		"77",
-		"77",
-		"77",
-		"396006",
-		"487",
-		"4059",
-		"45121",
-		"98",
-		"338863",
-		"1038807",
-	}
-	for i, expectation := range expectations {
-		sut := suts[i]
-		output := sut.SigFigs()
-		if output != expectation {
+	for _, sut := range suts {
+		var expr = process(sut.stimulus)
+		uut := expr.SigFigs()
+		if uut != sut.output.sigFigs {
 			t.Fail()
 		}
 	}
 }
 func Test_NonSigFigs(t *testing.T) {
-	var suts = act()
-	expectations := []string{
-		"99",
-		"",
-		"",
-		"190000.00",
-		"190000.00",
-		"190000.00",
-		"000",
-		"000",
-		"000",
-		"000",
-		"000",
-		"000",
-		"",
-	}
-	for i, expectation := range expectations {
-		sut := suts[i]
-		output := sut.NonSigFigs()
-		if output != expectation {
+	for _, sut := range suts {
+		var expr = process(sut.stimulus)
+		uut := expr.NonSigFigs()
+		if uut != sut.output.nonSigFigs {
 			t.Fail()
 		}
 	}
 }
 func Test_String(t *testing.T) {
-	var suts = act()
-	expectations := []string{
-		"0.00",
-		"0.02",
-		"(0.02)",
-		"77,190,000.00",
-		"77,190,000.00",
-		"(77,190,000.00)",
-		"396,006,000.00",
-		"487,000.00",
-		"4,059,000.00",
-		"45,121,000.00",
-		"98,000.00",
-		"(338,863,000.00)",
-		"1,038,807.00",
-	}
-	for i, expectation := range expectations {
-		sut := suts[i]
-		output := sut.String()
-		if output != expectation {
+	for _, sut := range suts {
+		var expr = process(sut.stimulus)
+		uut := expr.String()
+		if uut != sut.output.strOut {
 			t.Fail()
 		}
 	}
 }
 func Test_Head(t *testing.T) {
-	var suts = act()
-	expectations := []string{
-		"",
-		"",
-		"(",
-		"",
-		"",
-		"(",
-		"",
-		"",
-		"",
-		"",
-		"",
-		"(",
-		"",
-	}
-	for i, expectation := range expectations {
-		sut := suts[i]
-		output := sut.Head()
-		if output != expectation {
+	for _, sut := range suts {
+		var expr = process(sut.stimulus)
+		uut := expr.Head()
+		if uut != sut.output.head {
 			t.Fail()
 		}
 	}
 }
 func Test_Core(t *testing.T) {
-	var suts = act()
-	expectations := []string{
-		"0.",
-		"0.02",
-		"0.02",
-		"77",
-		"77",
-		"77",
-		"396,006",
-		"487",
-		"4,059",
-		"45,121",
-		"98",
-		"338,863",
-		"1,038,807.",
-	}
-	for i, expectation := range expectations {
-		sut := suts[i]
-		output := sut.Core()
-		if output != expectation {
+	for _, sut := range suts {
+		var expr = process(sut.stimulus)
```

Comment:
to clarify this even further:

`expr` is the "Software Under Test" or `sut`.  `suts` are actually a collection of test cases, which is a pair of one stimulus and one expectation.

My recommendation is this:
```
for _, case := range cases {
		var sut = process(case.stimulus)
		output := sut.Core()
		if output != case.expectation.core {
```

Reply:
addressed in bccb919. `testCase` and `testCases` used instead of `case` and `cases`, due to case being a reserved name in go